File: examples/snippets/modules/ROOT/examples/build-pekko-streamlets-scala/step3/src/main/scala/com/example/TestProcessor.scala
package com.example

import cloudflow.pekkostream._
import cloudflow.streamlets._
import cloudflow.streamlets.avro._
import cloudflow.pekkostream.scaladsl.RunnableGraphStreamletLogic

import org.apache.pekko.stream.scaladsl.Flow

//tag::processor[]
class TestProcessor extends PekkoStreamlet {
  val in                   = AvroInlet[Data]("in")
  val out                  = AvroOutlet[Data]("out", _.id.toString)
  final override val shape = StreamletShape.withInlets(in).withOutlets(out)

  val flow = Flow[Data].filter(_.id % 2 == 0)
  override final def createLogic = new RunnableGraphStreamletLogic() {
    def runnableGraph = plainSource(in).via(flow).to(plainSink(out))
  }
}
//end::processor[]

File: examples/snippets/modules/ROOT/examples/build-pekko-streamlets-java/step3/src/main/java/com/example/TestProcessor.java
package com.example;

import org.apache.pekko.NotUsed;
import org.apache.pekko.stream.*;
import org.apache.pekko.stream.javadsl.*;

import cloudflow.pekkostream.*;
import cloudflow.pekkostream.javadsl.*;
import cloudflow.streamlets.*;
import cloudflow.streamlets.avro.*;

//tag::processor[]
class TestProcessor extends PekkoStreamlet {
  AvroInlet<Data> inlet = AvroInlet.<Data>create("in", Data.class);
  AvroOutlet<Data> outlet = AvroOutlet.<Data>create("out", d -> Integer.toString(d.getId()), Data.class);

  public StreamletShape shape() {
    return StreamletShape.createWithInlets(inlet).withOutlets(outlet);
  }

  public PekkoStreamletLogic createLogic() {
    return new RunnableGraphStreamletLogic(getContext()) {
      public RunnableGraph<NotUsed> createRunnableGraph() {
        return getPlainSource(inlet)
          .via(Flow.<Data>create().filter(d -> d.getId() % 2 == 0))
          .to(getPlainSink(outlet));
      }
    };
  }
}
//end::processor[]